Sources/AppcuesKit/AppcuesKit.docc/2.0MigrationGuide.md

```diff
@@ -0,0 +1,54 @@
+| `ContainerDecoratingTrait`            | ``AppcuesContainerDecoratingTrait``                       |
+| `BackdropDecoratingTrait`             | ``AppcuesBackdropDecoratingTrait``                        |
+| `WrapperCreatingTrait`                | ``AppcuesWrapperCreatingTrait``                           |
+| `PresentingTrait`                     | ``AppcuesPresentingTrait``                                |
+| `ExperienceContainerViewController`   | ``AppcuesExperienceContainerViewController``              |
+| `ExperienceContainerLifecycleHandler` | ``AppcuesExperienceContainerEventHandler``                |
+| `PageMonitor`                         | ``AppcuesExperiencePageMonitor``                          |
+| `TraitError`                          | ``AppcuesTraitError``                                     |
+| `ExperienceTraitLevel`                | ``AppcuesExperiencePluginConfiguration/Level-swift.enum`` |
+
+### Changed
+
+- ``AppcuesExperienceTrait`` now has a new ``AppcuesExperienceTrait/metadataDelegate`` which supports sharing data between trait types and instances. See <doc:Traits> for more details.
+- The functionality of decorating traits set at the step level have been updated to apply only to the step on which they are set. New `undecorate()` functions are called to allow a trait to remove its decorations from the container or backdrop:
+    - ``AppcuesContainerDecoratingTrait`` added ``AppcuesContainerDecoratingTrait/undecorate(containerController:)``. This function should reverse the decoration applied in ``AppcuesContainerDecoratingTrait/decorate(containerController:)``
+    - ``AppcuesBackdropDecoratingTrait`` added ``AppcuesBackdropDecoratingTrait/undecorate(backdropView:)``. This function should reverse the decoration applied in ``AppcuesBackdropDecoratingTrait/decorate(backdropView:)``.
+- ``AppcuesContainerCreatingTrait`` changed it's  ``AppcuesContainerCreatingTrait/createContainer(for:with:)``, which now provides a configured instance of ``AppcuesExperiencePageMonitor`` instead of requiring the trait to create this object.
+    - ``AppcuesExperiencePageMonitor`` no longer has a public init `init(numberOfPages:currentPage:)`.
+- Appcues trait and action initialization has be revised with ``AppcuesExperiencePluginConfiguration`` supporting `Decodable` models for additional flexibility and robustness.
+    - For an ``AppcuesExperienceAction``, ``AppcuesExperienceAction/init(configuration:)`` provides a `configuration` object on which ``AppcuesExperiencePluginConfiguration/decode(_:)`` can be called with a `Decodable` type.
+    - For an ``AppcuesExperienceTrait``, ``AppcuesExperienceTrait/init(configuration:)`` provides a `configuration object` on which ``AppcuesExperiencePluginConfiguration/decode(_:)`` can be called with a `Decodable` type. The former `level` property provided in the trait init is now available from ``AppcuesExperiencePluginConfiguration/level-swift.property``. 
```

Comment:
`-swift.property` here, was `-swift.enum` in other spot? just checking if one of those might be incorrect